CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxTraitsFile.swift

```diff
@@ -24,10 +24,22 @@ let syntaxTraitsFile = SourceFileSyntax {
       """) {
       
       for child in trait.children {
-        VariableDeclSyntax("var \(raw: child.swiftName): \(raw: child.typeName)\(raw: child.isOptional ? "?" : "") { get }")
-        FunctionDeclSyntax("func with\(raw: child.name)(_ newChild: \(raw: child.typeName)\(raw: child.isOptional ? "?" : "")) -> Self")
+        VariableDeclSyntax("var \(raw: child.swiftName): \(raw: child.typeName)\(raw: child.isOptional ? "?" : "") { get set }")
       }
     }
+
+    ExtensionDeclSyntax("public extension \(trait.traitName)Syntax") {
+      FunctionDeclSyntax("""
+      /// Without this function, the `with` function defined on `SyntaxProtocol`
+      /// does not work on existentials of this protocol type.
+      @_disfavoredOverload
+      func with<T>(_ keyPath: WritableKeyPath<\(raw: trait.traitName)Syntax, T>, _ newChild: T) -> \(raw: trait.traitName)Syntax {
+        var copy: \(raw: trait.traitName)Syntax = self
+        copy[keyPath: keyPath] = newChild
```

Comment:
`layoutKeyPath` or something instead? `keyPath: keyPath` just looks weird 😅

Reply:
Well, it’s not just layouts right. `keyPath` can also point to trivia.